libmozevent/mercurial.py

```diff
@@ -154,10 +154,19 @@ def apply_build(self, build):
         except hglib.error.CommandError:
             raise Exception("Failed to update to revision {}".format(hg_base))
 
-        # Get current revision using full information tuple from hglib
-        revision = self.repo.identify(id=True).strip()
-        revision = self.repo.log(revision, limit=1)[0]
-        logger.info("Updated repo", revision=revision.node, repo=self.name)
+        # In this case revision is `hg_base`
+        logger.info("Updated repo", revision=hg_base, repo=self.name)
+
+        # See if the repo is clean
+        repo_status = self.repo.status()
+        if len(repo_status) != 0:
+            status = ""
```

Comment:
Let's log the entire `repo_status`, no need to parse it

Reply:
I don't think we want this, because each file from the repo is in that list with a status...

Reply:
Maybe we are interested only in modified files.